### PR TITLE
feat: Voice Message Sending

### PR DESCRIPTION
**Pull request by Icebluewolf**
## Summary
Adds support for sending voice messages. This includes the `waveform` and `duration_sec` parameters that can be used in the new VoiceMessage subclass.
Solves #2575 

Resolved, was a py-cord bug: ~~This will be a draft until I have confirmed the behavior of `waveform`s and `duration_sec` in interaction responses.
As of now I believe there is a discord bug that causes them to be ignored.~~
Resolved: ~~Also Pending: [DAPI #7158](https://github.com/discord/discord-api-docs/issues/7158)~~

## Information

- [ ] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed).
- [ ] This PR is **not** a code change (e.g. documentation, README, typehinting,
      examples, ...).

## Checklist

- [x] I have searched the open pull requests for duplicates.
- [x] If code changes were made then they have been tested.
  - [x] I have updated the documentation to reflect the changes.
- [x] If `type: ignore` comments were used, a comment is also left explaining why.
- [x] I have updated the changelog to include these changes.

### CODE REVIEW

**Comment by aaronjamt** [comment]
[DAPI #7158](https://github.com/discord/discord-api-docs/issues/7158) has been merged and I've been using this PR in a project for a couple weeks now without issues. Is it ready to merge?

**Comment by Icebluewolf** [comment]
As far as I know this has not been officially added to the API as it is not strictly documented. Also last time I tested it I was having issues getting the waveforms to display on interaction responses.

File: discord/file.py [hunk 1]
```diff
@@ -143,3 +147,59 @@ def close(self) -> None:
...
+
+
+class VoiceMessage(File):
+    """A special case of the File class that represents a voice message.
+
+    .. versionadded:: 2.7
+
+    .. note::
+
+        Just like File objects VoiceMessage objects are single use and are not meant to be reused in
```
Comment (Dorukyum) [inline discord/file.py:159]:
```suggestion
        Similar to File objects, VoiceMessage objects are single use and are not meant to be reused in
```

File: discord/file.py [hunk 2]
```diff
@@ -143,3 +147,59 @@ def close(self) -> None:
...
+        If this is not given then it defaults to ``fp.name`` or if ``fp`` is
+        a string then the ``filename`` will default to the string given.
+    description: Optional[:class:`str`]
+        The description of a file, used by Discord to display alternative text on images.
+    spoiler: :class:`bool`
+        Whether the attachment is a spoiler.
+    waveform: Optional[:class:`str`]
+        The base64 encoded bytearray representing a sampled waveform. Currently only for voice messages
+    duration_secs: Optional[:class:`float`]
+        The duration of the audio file. Currently only for voice messages
```
Comment (Dorukyum) [inline discord/file.py:187]: This is already in the VoiceMessage class so it should be clear. Also, why not just call this `duration`?
```suggestion
        The duration of the voice message.
```

File: discord/file.py [hunk 3]
```diff
@@ -143,3 +147,59 @@ def close(self) -> None:
...
+    filename: Optional[:class:`str`]
+        The filename to display when uploading to Discord.
+        If this is not given then it defaults to ``fp.name`` or if ``fp`` is
+        a string then the ``filename`` will default to the string given.
+    description: Optional[:class:`str`]
+        The description of a file, used by Discord to display alternative text on images.
+    spoiler: :class:`bool`
+        Whether the attachment is a spoiler.
+    waveform: Optional[:class:`str`]
+        The base64 encoded bytearray representing a sampled waveform. Currently only for voice messages
```
Comment (Dorukyum) [inline discord/file.py:185]: See comment below.
```suggestion
        The base64 encoded bytearray representing a sampled waveform.
```

File: discord/file.py [hunk 4]
```diff
@@ -143,3 +147,59 @@ def close(self) -> None:
...
+
+    __slots__ = (
+        "waveform",
+        "duration_secs",
+    )
+
+    def __init__(
+        self,
+        fp: str | bytes | os.PathLike | io.BufferedIOBase,
+        # filename: str | None = None,
```
Comment (Dorukyum) [inline discord/file.py:198]: Why

File: discord/file.py (hunk 2 above)
Comment (Icebluewolf) [reply discord/file.py:187]: That is what it is called in the Discord API

File: discord/file.py (hunk 4 above)
Comment (Icebluewolf) [reply discord/file.py]: Good catch, fixed

File: discord/file.py [hunk 5]
```diff
@@ -143,3 +147,59 @@ def close(self) -> None:
...
+    __slots__ = (
+        "waveform",
+        "duration_secs",
+    )
+
+    def __init__(
+        self,
+        fp: str | bytes | os.PathLike | io.BufferedIOBase,
+        filename: str | None = None,
+        waveform: str = "",
```
Comment (Dorukyum) [inline discord/file.py:199]: For consistency
```suggestion
        *,
        waveform: str = "",
```

**Review by Dorukyum — APPROVED** [review]

**Review by Lulalaby — APPROVED** [review]

**Pull request merged**